libs/dh/admin/feature-user-management/src/lib/overview/dh-users-overview-table.component.ts

```diff
@@ -94,40 +93,42 @@ export class DhUsersTabTableComponent implements AfterViewInit {
   dataSource = new WattTableDataSource<DhUser>();
   activeRow: DhUser | undefined = undefined;
 
-  @Input({ required: true }) set users(value: DhUsers) {
-    this.dataSource.data = value;
-  }
+  users = input.required<DhUsers>();
 
-  @Input({ required: true }) isLoading = false;
-  @Input({ required: true }) hasGeneralError = false;
+  isLoading = input.required<boolean>();
+  hasGeneralError = input.required<boolean>();
 
-  @Input({ required: true }) sortChanged!: (
-    prop: UserOverviewSortProperty,
-    direction: MarketParticipantSortDirctionType
-  ) => void;
+  sortChanged =
+    input.required<
+      (prop: UserOverviewSortProperty, direction: MarketParticipantSortDirctionType) => void
+    >();
 
-  @Output() reload = new EventEmitter<void>();
+  reload = output<void>();
 
-  @ViewChild(DhUserDrawerComponent)
-  drawer!: DhUserDrawerComponent;
+  private drawer = viewChild.required(DhUserDrawerComponent);
+  private usersTable = viewChild.required(WattTableComponent<DhUser>);
 
-  @ViewChild(WattTableComponent)
-  usersTable!: WattTableComponent<DhUser>;
+  constructor() {
+    effect(() => (this.dataSource.data = this.users()));
+  }
 
   ngAfterViewInit(): void {
-    this.usersTable.sortChange.pipe(takeUntilDestroyed(this._destroyRef)).subscribe((x) => {
-      const property = (x.active.charAt(0).toUpperCase() +
-        x.active.slice(1)) as UserOverviewSortProperty;
-      const direction = (x.direction.charAt(0).toUpperCase() +
-        x.direction.slice(1)) as MarketParticipantSortDirctionType;
-
-      this.sortChanged(property, direction);
-    });
+    this.usersTable()
```

Comment:
Question: Could this be moved into an effect. When we can drop the destoryRef thing.

Reply:
Apparently not. Moving the logic inside an `effect` fails with 

![image](https://github.com/user-attachments/assets/74ff4a23-3df4-4206-ab01-25c0500c5813)